### PR TITLE
scalacheck and partest

### PR DESCRIPTION
**Pull request by paulp**

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -182,6 +181,9 @@ PROPERTIES
   <property name="build-strap.dir" value="${build.dir}/strap"/>
   <property name="build-docs.dir" value="${build.dir}/scaladoc"/>
   <property name="build-libs.dir" value="${build.dir}/libs"/>
+  <property name="build-scalacheck.dir" value="${build.dir}/quick/classes/scalacheck"/>
+
+  <property name="scalacheck.jar" value="${build.dir}/pack/lib/scalacheck.jar"/>
```
Comment (jsuereth) [inline build.xml:186]: Is this used anywhere?

**Comment by jsuereth** [comment]
Not that you asked for a code review, but LGTM.

**Comment by jsuereth** [comment]
So, this failed the build failed with:

```

docs.partest:
    [mkdir] Created dir: /localhome/hudson/workspace/pr-scala-testsuite-linux-opt@2/build/scaladoc/scala-partest
 [scaladoc] Documenting 28 source files to /localhome/hudson/workspace/pr-scala-testsuite-linux-opt@2/build/scaladoc/scala-partest
 [scaladoc] /localhome/hudson/workspace/pr-scala-testsuite-linux-opt@2/src/partest/scala/tools/partest/RunScalacheck.scala:5: error: object scalacheck is not a member of package org
 [scaladoc] import org.scalacheck._
```

ALSO - Adding a hard dependency on scalacheck to partest is something we need to discuss.   If we do this, we also need to distribute the scalacheck jar we use in the Scala distribution, so those few folk who use partest outside of Scala don't become broken.   I'm not saying we shouldn't do this, just that we should have a chat about it and possible alternatives.

And yes, people use partest outside of the scala distribution for testing plugins.   

Is there a way I can pull in https://github.com/paulp/scala/commit/56f08a3234f6ac7a868faff42f94bffccfd1ee68 in its own pull request?

**Comment by paulp** [comment]
You have my permission to cherry-pick it.

**Comment by paulp** [comment]
Or never mind, it looks like it's in https://github.com/scala/scala/pull/717 .

File: src/library/scala/collection/concurrent/TrieMap.scala [hunk 2]
```diff
@@ -473,7 +473,7 @@ extends CNodeBase[K, V] {
   private def computeSize(ct: TrieMap[K, V]): Int = {
     var i = 0
     var sz = 0
-    val offset = math.abs(util.Random.nextInt()) % array.length
+    val offset = if (array.length == 0) 0 else math.abs(util.Random.nextInt()) % array.length
```
Comment (axel22) [inline src/library/scala/collection/concurrent/TrieMap.scala:476]: Modulo could be replaced with `util.Random.nextInt(array.length)`, as Rudiger Keller suggested.

**Comment by jsuereth** [comment]
Paul,  I may close this on monday and "finish" the publishing portions of it.  I'll then re-issue with the fixes.   We can discuss the merits of publishing our own scalacheck then.   I can at least get the build working again.

**Comment by jsuereth** [comment]
Closing this since it doesn't merge.    I'm working on this locally, will re-submit as two pull requests, if necessary.   I'm thinking it might be time for scalacheck to exist as it's own test runner....

**Pull request closed without merging**